src/com/google/javascript/jscomp/Es6ConvertSuper.java

```diff
@@ -108,44 +108,59 @@ private boolean isInterface(Node classNode) {
   }
 
   private void visitSuper(Node node, Node parent) {
-    Node enclosingCall = parent;
-    Node potentialCallee = node;
-    if (!parent.isCall()) {
-      enclosingCall = parent.getParent();
-      potentialCallee = parent;
+    Node exprRoot = node;
+    // properties with getElems aren't considered qualified names, so we have
+    // to do this without using any qName helpers.
+    for (;
+        exprRoot.getParent().isGetElem() || exprRoot.getParent().isGetProp();
+        exprRoot = exprRoot.getParent()) {
+      if (exprRoot.getParent().getFirstChild() != exprRoot) {
```

Comment:
I think this is going too far.
If you have `super.foo.bar.baz()`, we only need to go as far as `super.foo` to determine what to do.